src/container_support/training.py

```diff
@@ -40,7 +52,8 @@ def start(cls):
             message = 'uncaught exception during training: {}\n{}\n'.format(e, trc)
             logger.error(message)
             TrainingEnvironment.write_failure_file(message, base_dir)
-            exit_code = 1 if not hasattr(e, 'errno') else e.errno
+            error_number = DEFAULT_FAILURE_CODE if not hasattr(e, 'errno') else e.errno
```

Comment:
I would flip this so you don't have the `if not` + `else`, i.e. `e.errno if hasattr(e, 'errno') else DEFAULT_FAILURE_CODE`